tests/test_pages.py

```diff
@@ -60,22 +61,39 @@ def chrome_options(chrome_options):
     return chrome_options
 
 
-@pytest.fixture
-def user(variables):
+def _user(variables):
     return User(variables.get('username', 'admin'), variables.get('password', 'changeme'),
                 variables.get('name', 'Admin User'))
 
 
-@pytest.fixture(params=PAGES)
-def url(request, base_url):
-    if not base_url:
-        raise ValueError('Base URL not specified')
-    return base_url + request.param
+@pytest.fixture(scope='session')
+def user(variables):
+    return _user(variables)
+
+
+def pytest_generate_tests(metafunc):
+    if 'url' in metafunc.fixturenames:
+        base_url = metafunc.config.option.base_url
+        if not base_url:
```

Comment:
why the `if` instead of just calling `assert` directly?

Reply:
Good point